src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -204,29 +207,79 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
                 return false;
             }
 
+            // Clear the header if status line is recieved again. This signifies that there are multiple response headers (like in redirection).
+            response.Headers.Clear();
+            
+            response.Content.Headers.Clear();        
+
+            int responseHeaderLength = responseHeader.Length;
+
             // Check if line begins with HTTP/1.1 or HTTP/1.0
             int prefixLength = s_httpPrefix.Length;
-            if ((responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
+            int versionIndex = prefixLength + 2;
+
+            if ((versionIndex < responseHeaderLength) && (responseHeader[prefixLength] == '1') && (responseHeader[prefixLength + 1] == '.'))
             {
-                if (responseHeader[prefixLength + 2] == '1')
+                if (responseHeader[versionIndex] == '1')
                 {
-                    response.Version = new Version(1, 1);
+                    response.Version = HttpVersion.Version11;
                 }
-                else if (responseHeader[prefixLength + 2] == '0')
+                else if (responseHeader[versionIndex] == '0')
                 {
-                    response.Version = new Version(1, 0);
+                    response.Version = HttpVersion.Version10;
                 }
             }
-
-            // Parse first 3 characters after a space as status code
+         
             // TODO: Parsing errors are treated as fatal. Find right behaviour
-            int statusCodeLength = 3;
-            int codeIndex = responseHeader.IndexOf(' ') + 1;
-            string strStatusCode = responseHeader.Substring(codeIndex, statusCodeLength);
-            response.StatusCode = (HttpStatusCode)(int.Parse(strStatusCode));
+           
+            int spaceIndex = responseHeader.IndexOf(SpaceChar);
+
+            if (spaceIndex > -1)
+            {
+                int codeStartIndex = spaceIndex + 1;
+                int codeEndIndex = codeStartIndex + StatusCodeLength;
```

Comment:
This declaration of codeEndIndex can move into the if block just before it's used.